### PR TITLE
#167165058 Reset Password feature

### PR DESCRIPTION
**Pull request by Tunji17**
### What does this PR do?
User should be able to reset password

#### Description of Task to be completed?
wrote an integration test for forgot password and reset password
wrote forgot password and reset password controllers
adds a helper function to interface with the postal of email
adds a helper function to verify jw token
adds routes for both controllers

#### How should this be manually tested?

> 1. Fetch this branch on the terminal using 
> `git fetch origin ft-resetpasswordfeature-167165058`
> 2. Install dependencies using `npm install`
> 3. Start the server with `npm run server`
> 4. On Postman, sign up by making a POST request to /api/v1/auth/signup.
> 5. On Postman, make a POST request to /api/v1/auth/forgotpassword.
> 5. Check your inbox of the email you signed up with for a reset password link from authors haven.
> 6. Upon redirection to the browser, copy the link in the URL bar.
> 7. Paste this link in Postman and make a PATCH request.
> 

#### Any background context you want to provide?
User cannot reset the password with the same token more than once

#### What are the relevant pivotal tracker stories?
[#167165058](https://www.pivotaltracker.com/story/show/167165058)

### CODE REVIEW

File: controllers/AuthController.js [hunk 1]
```diff
@@ -211,6 +213,7 @@ export default class AuthController {
         },
       });
     } catch (error) {
+      /* istanbul ignore next-line */
```
Comment (mazma1) [inline controllers/AuthController.js:216]: Why don't you want this line covered?

File: middlewares/inputValidation.js [hunk 2]
```diff
@@ -175,5 +175,53 @@ const validate = {
...
+      .withMessage('Password must be at least 8 characters'),
+    (req, res, next) => {
+      const errors = validationResult(req);
+      const errorMessage = {};
+      if (!errors.isEmpty()) {
+        /* istanbul ignore next-line */
+        errors.array({ onlyFirstError: true }).forEach((error) => {
+          errorMessage[error.param] = error.msg;
+        });
+        /* istanbul ignore next-line */
```
Comment (mazma1) [inline middlewares/inputValidation.js]: Who introduced this?

File: controllers/AuthController.js (hunk 1 above)
Comment (Tunji17) [reply controllers/AuthController.js:216]: It's the social login feature, my feature is properly covered but passport and that line was dropping my coverage so ignored it. i didn't collaborate fully with him on that feature so i don't know what to test for.

File: middlewares/inputValidation.js (hunk 2 above)
Comment (Tunji17) [reply middlewares/inputValidation.js]: @mazma1 I just saw that I introduced it and I have pushed a fix to remove it and write a test for it

**Pull request merged**